src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -256,6 +256,62 @@ export class ExposureNotificationService {
     await this.recordKeySubmission();
   }
 
+  calculateNeedsSubmission(exposureStatus: ExposureStatus, today: Date): boolean {
+    if (exposureStatus === null || today === null) return false;
+    if (exposureStatus.type !== ExposureStatusType.Diagnosed) return false;
+
+    if (isNaN(today.getTime())) return true;
+
+    const cycleEndsAt = new Date(exposureStatus.cycleEndsAt);
+    if (isNaN(cycleEndsAt.getTime()) === false) {
+      // We're done submitting keys
+      // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+      if (daysBetweenUTC(today, cycleEndsAt) <= 0) return false;
+    }
+
+    const submissionLastCompletedAt = exposureStatus.submissionLastCompletedAt;
+    if (!submissionLastCompletedAt) {
+      return true;
+    }
+
+    const lastSubmittedDay = new Date(submissionLastCompletedAt);
+
+    if (isNaN(lastSubmittedDay.getTime())) return true;
+
+    // This has to be based on UTC timezone https://github.com/cds-snc/covid-shield-mobile/issues/676
+    return daysBetweenUTC(lastSubmittedDay, today) > 0;
+  }
+
+  public updateExposure(exposureStatus: ExposureStatus, today: Date): ExposureStatus {
+    if (exposureStatus.type === ExposureStatusType.Monitoring) return exposureStatus;
+    switch (exposureStatus.type) {
+      case ExposureStatusType.Diagnosed:
+        // There is a case where using UTC and device timezone could mess up user experience. See `date-fn.spec.ts`
+        // Let's use device timezone for resetting exposureStatus for now
+        // Ref https://github.com/cds-snc/covid-shield-mobile/issues/676
+        if (daysBetween(today, new Date(exposureStatus.cycleEndsAt)) <= 0) {
+          return {type: ExposureStatusType.Monitoring, lastChecked: exposureStatus.lastChecked};
+        } else {
+          return {
+            ...exposureStatus,
+            needsSubmission: this.calculateNeedsSubmission(exposureStatus, today),
+          };
+        }
+      case ExposureStatusType.Exposed:
+        if (
+          daysBetween(new Date(exposureStatus.summary.lastExposureTimestamp || today.getTime()), today) >=
```

Comment:
I find this line a bit hard to read. Would you be okay with just defining the variable in the case block like the old code does?
```
const lastExposureAt = new Date(currentStatus.summary.lastExposureTimestamp || today.getTime());
```

That would require adding a lint exception but is worth it IMHO.

Reply:
It's a `const` that is only used once, so storing it in a variable isn't really necessary. Perhaps there's another way to make it easier to read.